nmrfx-structure/src/main/java/org/nmrfx/structure/noe/NOECalibrator.java

```diff
@@ -422,10 +415,25 @@ public void findRedundant() {
         }
     }
 
+    /**
+     * Checks if the entity is a residue and returns it as a residue, if not a message is logged and null is returned.
+     * @param entity The entity to check
+     * @return The entity cast to a residue
+     */
+    private Residue checkEntityIsResidue(Entity entity) {
+        Residue residue = null;
+        if (entity instanceof Residue ) {
+            residue = (Residue) entity;
+        } else {
+            log.info("Entity is not a polymer: {}", entity.getName());
+        }
+        return residue;
```

Comment:
Minor, but this could be slightly simplified:

```suggestion
        if (entity instanceof Residue) {
            return (Residue) entity;
        } else {
            log.info("Entity is not a polymer: {}", entity.getName());
            return null;
        }
```

That way it's clearer that the intermediate variable is never accessed.